src/GraphQL/Extensions/ObjectExtensions.cs

```diff
@@ -272,6 +272,7 @@ public MemberFieldInfo(string key, MemberInfo member, bool isInitOnly, bool isRe
     /// will have 'FirstName' key but its value should be set to 'FName' property of created object.
     /// </param>
     /// <remarks>There is special handling for strings, IEnumerable&lt;T&gt;, Nullable&lt;T&gt;, and Enum.</remarks>
+    [RequiresUnreferencedCode("Recursively converts propertyValue to fieldType.")]
     public static object? GetPropertyValue(this object? propertyValue, Type fieldType, IGraphType mappedType)
```

Comment:
One of the benefits of this new design is that the amount of code within `ObjectExtensions.GetPropertyValue` is notably simplified.  Also the similar code within `ObjectExtensions.Compiled.cs` is greatly simplified.  In general, all the reflection-based code got moved to `CustomListConverterFactory` (where it is much easier understood), and all the code dedicated specifically to array or `List<T>` generation was removed as well, being replaced with `ArrayListConverterFactory` and `DefaultListConverterFactory`, respectively.  Finally, the type analysis code was also removed and replaced with a dictionary lookup within `ValueConverter.GetListConverterFactory` and the generic type analysis within `ListConverterFactoryBase`.

In short, the design is better regardless of whether it allows for custom list converters or not.